inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+from typing import List, Literal, Optional, Type, Union
```

Comment:
Q: What is designed usage pattern for the block? To "ground" SAM with bboxes predicted by another model I guess. From what I see the classes of original boxes are to be preserved (if present) or default ones to be added by numbers. The latter approach is probably unstable for sequences processing

Reply:
>Q: What is designed usage pattern for the block? To "ground" SAM with bboxes predicted by another model I guess.

yes

Reply:
I think it's stable because when you don't prompt you will only get one mask, If you prompt with multiple boxes, it will assign class based on the class from the box that was prompt for that mask